sys/geom/geom_ctl.h

```diff
@@ -41,15 +41,26 @@
  */
 #define GCTL_VERSION	2
 
+#ifdef _KERNEL
+struct gctl_kreq_arg {
```

Comment:
This seems unnecessary, it's the same as `gctl_req_arg` in the kernel other than dropping `nlen`?

Reply:
I agree, these don't seem necessary.

Reply:
Originally I was thinking of upstreaming these changes as I really don't like exposing kernel fields in user structures, etc.  However, this was an earlier patch in the series and as I got farther along I tended to just do unions for kernel pointers, so I will simplify the gctl bits (though I still think overloading the same structure for both is pretty gross, and there are many unused fields to boot)